### DISCUSSION TITLE
Importing a grid from an hydro simulation

### DISCUSSION THREAD

**Discussion by Rogerod23** [dullemond/radmc3d-2.0#44] [Q&A]
Hello all, I'm trying to import a grid that I have in spherical coordinates that was generated using an HD code. 
The grid goes from  0.02 to 110 pc in the radial coordinate (r) and from 0 to pi in the angular coordinate (theta). The simulation was run with symmetry around the phi axis. 
The problem is that when I try to run a test with radmc3 to calculate the temperature with mctherm I get the following text:
ERROR: Theta grid crosses midplane, but there is no
        interface at pi/2. Maybe check the precision 
        of the theta-grid: the equator interface must
        be really exactly pi/2 to 12 digits or more.
Is it because I'm supposed to exactly define my grid from 0 to pi/2? Or is it because radmc3d is assuming that my grid has a mirroring effect? If so, how could I change this? 
 I have attached my problem setup, this is just one of the .py setups from the examples directory that has been edited to import my r, theta coordinates. 

Bests,

-Rogelio.
[problem_setup.py.txt](https://github.com/dullemond/radmc3d-2.0/files/12555771/problem_setup.py.txt)

**Comment by birnstiel** [comment]
Not sure why RADMC3D needs this but it tells you what to do: it wants an interface at pi/2. I haven't checked your grid, but either you have already a grid interface there and perhaps you just don't write it out with enough precision (which happened to me a bunch of times), or you do not have an interface at pi/2, then you need to create one, i.e. divide the cell that crosses pi/2 into two cells with the interface between them at pi/2. Hope that helps.
